src/taipy/core/submission/submission.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import uuid
+from datetime import datetime
+from typing import List, Optional, Union
+
+from .._entity._entity import _Entity
+from .._entity._labeled import _Labeled
+from .._entity._reload import _self_reload, _self_setter
+from ..job._job_manager_factory import _JobManagerFactory
+from ..job.job import Job, JobId
+from .submission_id import SubmissionId
+from .submission_status import SubmissionStatus
+
+
+class Submission(_Entity, _Labeled):
+    """Hold a user function that will be executed, its parameters and the results.
+
+    A `Task` brings together the user code as function, the inputs and the outputs as data nodes
+    (instances of the `DataNode^` class).
+
+    Attributes:
+        entity_id (str): The identifier of the `Submittable^` entity.
+        jobs (Union[List[JobId], List[Job]]): A list of jobs.
+        function (callable): The python function to execute. The _function_ must take as parameter the
+            data referenced by inputs data nodes, and must return the data referenced by outputs data nodes.
+        input (Union[DataNode^, List[DataNode^]]): The list of inputs.
+        output (Union[DataNode^, List[DataNode^]]): The list of outputs.
+        id (str): The unique identifier of the task.
+        owner_id (str):  The identifier of the owner (sequence_id, scenario_id, cycle_id) or None.
+        parent_ids (Optional[Set[str]]): The set of identifiers of the parent sequences.
+        version (str): The string indicates the application version of the task to instantiate. If not provided, the
+            latest version is used.
+        skippable (bool): If True, indicates that the task can be skipped if no change has been made on inputs. The
+            default value is _False_.
+
+    """
+
+    _ID_PREFIX = "SUBMISSION"
+    _MANAGER_NAME = "submission"
+    __SEPARATOR = "_"
+
+    def __init__(
+        self,
+        entity_id: str,
+        jobs: Optional[Union[List[JobId], List[Job]]] = None,
+        submission_id: Optional[str] = None,
+    ):
```

Comment:
My thinking for it is that, we will create a Submission object before creating a job, then we will create jobs and propagate jobs to the submission object after that. So based on that flow, I think not having them in the Submission signature makes sense. But it can also be easier in different cases to have them in the signature as well, though don't think it's accurate for the use case

Reply:
Without the creation_date and the status in the signature, when reading a submission from our repo, we will first instantiate a submission with `status` equals to `UNDEFINED` and `creation_date` equals to `datetime.now()`. Immediately after, we will set the proper values. This is strange to me.
Plus it is not consistent with other entities.